scalameta/contrib/shared/src/main/scala/scala/meta/contrib/ScaladocToken.scala

```diff
@@ -0,0 +1,197 @@
+package scala.meta.contrib
+
+/** Represents a scaladoc element */
+sealed abstract class ScaladocToken
+
+/**
+ * The available tokens and their documentation are obtained from:
+ * @see http://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html
+ */
+object ScaladocToken {
+
+  /** A blank line */
+  final case object Break extends ScaladocToken
+
+  /** A paragraph which failed parsing */
+  final case class Unknown(text: String) extends ScaladocToken
+
+  // Text
+
+  /** An inline part of a text block */
+  trait TextPart {
```

Comment:
i didn't implement the reference type as a subclass of this. primarily because I'm not sure what the syntax is (some examples on the web include multiple words between the bracket pairs).

another aspect i sort of struggled with... in my case, the goal is to be able to format this, but this is not the place to decide on formatting, so i only added this `syntax` method to inline tokens, those which don't span multiple lines, to reduce the number of case expressions in the match statement in the formatter, listing only a subset of all possible tokens.

so if reference could potentially be split across lines, with multiple words, this complicates things.

there are also other examples, such as markdown of italics, fixed-width font, etc. they are not well documented, so I skipped them, as well.